lib/bolt/cli.rb

```diff
@@ -676,18 +689,27 @@ def initialize_project
 
       # Warn the user if the project directory already exists. We don't error here since users
       # might not have installed any modules yet.
+      # If both bolt.yaml and bolt-project.yaml exist, this will just warn
+      # about bolt-project.yaml and subsequent Bolt actions will warn about
+      # both files existing
       if config.exist?
-        @logger.warn "Found existing project directory at #{project}"
+        @logger.warn "Found existing project directory at #{project}. Skipping file creation."
+      # This won't get called if bolt-project.yaml exists
+      elsif old_config.exist?
+        @logger.warn "Found existing #{old_config.basename} at #{project}. "\
+        "Attempting to update to #{config.basename}."
+        update_project_config(old_config)
```

Comment:
This seems more like `bolt project migrate` type behavior... I don't have strong feelings about it though.

Reply:
It definitely is, I wasn't sure about adding a whole other subcommand as part of this though. and it feels like it follows the principle of "I'm going to tell you this is wrong and how to fix it" instead of "I can tell this is wrong so I'm going to fix it if I can".